Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -537,275 +519,197 @@ - (void)messageComposeViewController:(MFMessageComposeViewController *)controlle
-        [self sendTextToPhoneNumber:phoneNumber];
-    } else if (indexPath.section == MessageComposeTableViewControllerSectionInviteFlow) {
-        void (^showInvite)() = ^{
-            OWSInviteFlow *inviteFlow =
-                [[OWSInviteFlow alloc] initWithPresentingViewController:self contactsManager:self.contactsManager];
-            [self presentViewController:inviteFlow.actionSheetController
-                               animated:YES
-                             completion:^{
-                                 [self.tableView deselectRowAtIndexPath:indexPath animated:YES];
-                             }];
-        };
-
-        if (self.presentedViewController) {
-            // If search controller is active, dismiss it first.
-            [self dismissViewControllerAnimated:YES completion:showInvite];
-        } else {
-            showInvite();
-        }
-    } else {
-        OWSAssert(indexPath.section == MessageComposeTableViewControllerSectionContacts)
+#pragma mark - UISearchBarDelegate
 
-            if (!self.searchController.active && self.hasNoContacts)
-        {
-            // TODO: We could do something here, like show the "invite contacts" UI.
-            return;
-        }
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText
+{
+    [self searchTextDidChange];
+}
 
-        Contact *contact = [self contactForIndexPath:indexPath];
-        NSString *contactIdentifier = [[contact textSecureIdentifiers] firstObject];
-        [self dismissViewControllerAnimated:YES
-                                 completion:^() {
-                                     [Environment messageIdentifier:contactIdentifier withCompose:YES];
-                                 }];
-    }
+- (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar
+{
+    [self searchTextDidChange];
 }
 
-- (Contact *)contactForIndexPath:(NSIndexPath *)indexPath {
-    Contact *contact = nil;
+- (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar
+{
+    [self searchTextDidChange];
+}
 
-    if (self.searchController.active) {
-        contact = [self.searchResults objectAtIndex:(NSUInteger)indexPath.row];
-    } else {
-        contact = [self.contacts objectAtIndex:(NSUInteger)indexPath.row];
-    }
+- (void)searchBarResultsListButtonClicked:(UISearchBar *)searchBar
+{
+    [self searchTextDidChange];
+}
 
-    return contact;
+- (void)searchBar:(UISearchBar *)searchBar selectedScopeButtonIndexDidChange:(NSInteger)selectedScope
+{
+    [self searchTextDidChange];
 }
 
-- (void)updateContacts {
-    OWSAssert([NSThread isMainThread]);
+- (void)searchTextDidChange
+{
+    [self updateSearchPhoneNumbers];
 
-    self.contacts = [self filteredContacts];
-    [self updateSearchResultsForSearchController:self.searchController];
-    [self.tableView reloadData];
-    // TODO revisit this after https://github.com/WhisperSystems/Signal-iOS/pull/2058 is merged
-    [self showEmptyBackgroundViewIfNecessary];
```

Comment:
@michaelkirk I think you added `showEmptyBackgroundViewIfNecessary` recently - I've removed it.  Is it clear why?